docs/thrust/cmake_options.rst

```diff
@@ -69,6 +69,18 @@ Generic CMake Options
    -  If true, installation rules will be generated for thrust. Default
       is ``ON``.
 
+-  ``THRUST_DISPATCH_TYPE={Dynamic, Force32bit, Force64bit}``
+
+   -  Allows the user to force Thrust to use a specific size for the offset type. Default
+      is ``Dynamic``.
+
+      -  ``Dynamic`` lets Thrust choose the index type based on input size, allowing
+         large inputs and optimal performance at the cost of increased compile time and binary size
+      -  ``Force32bit`` forces Thrust to use a 32 bit offset type. This improves compile time and
+         binary size but limits the input size.
+      -  ``Force64bit`` forces Thrust to use a 64bit offset type. This improves compile time and
+         binary size and allows large input sizes. However, it might degrade runtime performance
```

Comment:
It varies a lot depending on the algorithm. This [Google Sheet](https://docs.google.com/spreadsheets/d/18er8G8JfDbYOj-awIbHkXecCVQ2EkdqCESOvndCji30/edit?usp=sharing) summarizes benchmark results for some algorithms. Note, it's limited to algorithms which, today, do not yet support larger-than-`INT_MAX` number of items in CUB, as that was the focus of the research. I wouldn't draw conclusions to other algorithms. As a rule of thumb, algorithms that use decoupled look-back and carry the offset type (e.g., `copy_if`) are more performance-sensitive. Some others don't see a noticeable performance change at all (e.g., `scan`)

Reply:
I would not want to put any real benchmark data in it, because that will always be moot.

If we do not commit to keep that data on trak with cccl ToT and the latests architectures, it is imho better to just note this and let the user decide

Reply:
Do we have a list of algorithms that we can switch to 64 bit only without affecting performance?

Reply:
Thanks. To be clear, I don't want to include numbers in the docs here. I just want to know _generally_ how sensitive Thrust is to this choice. Yes, it looks like some algorithms are unaffected. It would be very interesting to see what happens to libcudf compile times / binary size if we always dispatched the unaffected algorithms to 64 bit offsets, and compiled two kernels (32 and 64) for the algorithms where it matters for performance.